sw/device/tests/sim_dv/otp_ctrl_lc_signals_test.c

```diff
@@ -0,0 +1,378 @@
+ * the data in the buffer. If exp_result is set to kExpectPassed, the test
+ * expects the test to pass without errors and the read data to match with the
+ * data in the buffer.
+ */
+static void otp_read_test(int32_t address, const uint8_t *buffer, uint32_t size,
+                          exp_test_result_t exp_result) {
+  otp_ctrl_testutils_wait_for_dai(&otp);
+  for (int i = address; i < address + size; i += sizeof(uint64_t)) {
+    uint64_t act, exp;
+    CHECK_DIF_OK(
+        dif_otp_ctrl_dai_read_start(&otp, kDifOtpCtrlPartitionSecret2, i));
+    otp_ctrl_testutils_wait_for_dai(&otp);
+    check_dai_access_error(address, exp_result);
+
+    // Check whether the read data matches.
+    memcpy(&exp, &buffer[i], sizeof(exp));
+    CHECK_DIF_OK(dif_otp_ctrl_dai_read64_end(&otp, &act));
+    if (exp_result == kExpectFailed) {
+      CHECK(act != exp,
+            "0x%x == 0x%x (act == exp) is not expected at address 0x%x", act,
+            exp, address);
+    } else {
+      CHECK(act == exp,
+            "0x%x != 0x%x (act != exp) is not expected at address 0x%x", act,
+            exp, address);
+    }
+  }
+}
+
+/**
+ * Test wrapper function that first calls the write and then the read test.
+ */
+static void otp_write_read_test(test_mode_t test_mode, int32_t address,
+                                const uint8_t *buffer, uint32_t size,
+                                exp_test_result_t exp_result) {
+  if (test_mode == kWriteMode || test_mode == kWriteReadMode) {
+    LOG_INFO("Write test at 0x%x...", address);
+    otp_write_test(address, buffer, size, exp_result);
+  }
+  if (test_mode == kReadMode || test_mode == kWriteReadMode) {
+    LOG_INFO("Read test at 0x%x...", address);
+    otp_read_test(address, buffer, size, exp_result);
+  }
+}
+
+/**
+ * Test wrapper function that calls the write and read tests above for all
+ * objects within the SECRET2 partition.
+ */
+static void run_otp_access_tests(test_mode_t test_mode,
+                                 exp_test_result_t exp_result) {
+  otp_write_read_test(
+      test_mode,
+      OTP_CTRL_PARAM_RMA_TOKEN_OFFSET - OTP_CTRL_PARAM_SECRET2_OFFSET,
+      (uint8_t *)kOtpRmaToken, OTP_CTRL_PARAM_RMA_TOKEN_SIZE, exp_result);
+  otp_write_read_test(test_mode,
+                      OTP_CTRL_PARAM_CREATOR_ROOT_KEY_SHARE0_OFFSET -
+                          OTP_CTRL_PARAM_SECRET2_OFFSET,
+                      (uint8_t *)kOtpRootKeyShare0,
+                      OTP_CTRL_PARAM_CREATOR_ROOT_KEY_SHARE0_SIZE, exp_result);
+  otp_write_read_test(test_mode,
+                      OTP_CTRL_PARAM_CREATOR_ROOT_KEY_SHARE1_OFFSET -
+                          OTP_CTRL_PARAM_SECRET2_OFFSET,
+                      (uint8_t *)kOtpRootKeyShare1,
+                      OTP_CTRL_PARAM_CREATOR_ROOT_KEY_SHARE1_SIZE, exp_result);
+}
+
+/**
+ * Initialize and advance to kDifKeymgrStateCreatorRootKey state.
+ */
+static void keymgr_advance_to_creator_root_key() {
```

Comment:
nit: From C style conventions, add `void` param to this function and other functions in this file:

```c
static void keymgr_advance_to_creator_root_key(void)
```

Reply:
indeed, forgot to do that - will add it.